app/assets/javascripts/assignment_swapping.js

```diff
@@ -0,0 +1,57 @@
+function add_events()
+{
+  var form = document.getElementById("swapping_form");
+
+  if(form != null)
+  {
+   form.addEventListener("submit", function(event){ check_valid_form(event); })
+  }
+
+  var input_elements = document.getElementsByTagName("input");
+
+  for(var i = 0; i<input_elements.length; i++)
+  {
+    if(input_elements[i].type == "checkbox")
+    {
+      input_elements[i].addEventListener("change", function(){document.getElementById("submit_btn").disabled = false;})
```

Comment:
This seems wrong and should probably just use `check_valid_form` as the callback. Regardless, this shouldn't manage the disabled state directly (it doesn't know if the button should be disabled just that an input changed).